src/main/java/core/basesyntax/service/DataConverterImpl.java

```
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        if (lines == null) {
+            throw new RuntimeException("Input list cannot be null");
+        }
+        for (int i = 1; i < lines.size(); i++) {
+            String[] data = lines.get(i).split(",");
+            FruitTransactionService transactionService = new FruitTransactionServiceImpl();
+            transactionService.createTransaction(Operation.fromCode(data[0].trim()),
+                    data[1].trim(),
+                    Integer.parseInt(data[2].trim()));
```

Parsing and validation happen without guarding parse errors: `Integer.parseInt(data[2].trim())` is called directly. If the quantity is not a valid integer or is negative, this will throw NumberFormatException or silently accept negatives. The checklist requires robust input validation and informative runtime exceptions (`Don't ignore exceptions in try-catch blocks`). Wrap parsing in try/catch and verify quantity >= 0; on error throw a RuntimeException that includes line number and line content.